flask_hal/link.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+"""
+flask_hal.link
+==============
+
+Implements the ``HAL`` Link specification.
+"""
+
+# Standard Libs
+import json
+
+
+VALID_LINK_ATTRS = [
+    'name',
+    'title',
+    'type',
+    'deprecation',
+    'profile',
+    'templated',
+    'hreflang'
+]
+
+
+class Collection(object):
```

Comment:
I played with list inheritance a get this result which is way easier and uses list functions as well.. 

``` python
import json

class Dog(object):

    def __init__(self, name):
        self.name = name


class MyList(list):

    def to_json(self):
        return json.dumps([obj.__dict__ for obj in self])

dogs = MyList()
dogs.append(Dog('alik'))

print dogs.to_json()   # [{"name": "alik"}]

```